### ISSUE TITLE
Parse always add time info if grammar error

### ISSUE THREAD

**Issue by LuisMorenoSerrano** [tree-sitter/tree-sitter#3232]
### Problem

`tree-sitter parse` and a file with a grammar error always adds a line at the end that shows the same information as if the `--time` flag had been activated.

This behavior is caused by:

### tree-sitter/cli/src/parse.rs
```
...
if first_error.is_some() || opts.print_time {
...
```
Why show execution time when an error occurs, even if the corresponding flag `--time` has not been specified?

### Expected behavior

I think it is not a good idea to mix both situations in the same piece of code, or at least redirect this kind of error info via `stderr` instead of `stdout`.

**Comment by amaanq** [comment]
See my comment here https://github.com/tree-sitter/tree-sitter/pull/3260#issuecomment-2040825906

tl;dr: it's useful to know where parsing went wrong (the error node info at the end), that's why it's printed, and removing just the time aspect there and only adding it when `-t` is specified is awkward
